include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -1246,6 +1301,273 @@ TensorProductMatrixSymmetricSumCollection<dim, Number, n_rows_1d>::
 
 
 
+namespace TensorProductMatrixCreator
+{
+  namespace internal
+  {
+    template <typename Number>
+    std::tuple<FullMatrix<Number>, FullMatrix<Number>, bool>
+    create_referece_mass_and_striffness_matrices(
+      const FiniteElement<1> &fe,
+      const Quadrature<1> &   quadrature)
+    {
+      Triangulation<1> tria;
+      GridGenerator::hyper_cube(tria);
+
+      DoFHandler<1> dof_handler(tria);
+      dof_handler.distribute_dofs(fe);
+
+      MappingQ1<1> mapping;
+
+      const unsigned int n_dofs_1D = fe.n_dofs_per_cell();
+
+      FullMatrix<Number> mass_matrix_reference(n_dofs_1D, n_dofs_1D);
+      FullMatrix<Number> derivative_matrix_reference(n_dofs_1D, n_dofs_1D);
+
+      FEValues<1> fe_values(mapping,
+                            fe,
+                            quadrature,
+                            update_values | update_gradients |
+                              update_JxW_values);
+
+      fe_values.reinit(tria.begin());
+
+      const auto lexicographic_to_hierarchic_numbering =
+        Utilities::invert_permutation(
+          FETools::hierarchic_to_lexicographic_numbering<1>(
+            fe.tensor_degree()));
+
+      for (const unsigned int q_index : fe_values.quadrature_point_indices())
+        for (const unsigned int i : fe_values.dof_indices())
+          for (const unsigned int j : fe_values.dof_indices())
+            {
+              mass_matrix_reference(i, j) +=
+                (fe_values.shape_value(lexicographic_to_hierarchic_numbering[i],
+                                       q_index) *
+                 fe_values.shape_value(lexicographic_to_hierarchic_numbering[j],
+                                       q_index) *
+                 fe_values.JxW(q_index));
+
+              derivative_matrix_reference(i, j) +=
+                (fe_values.shape_grad(lexicographic_to_hierarchic_numbering[i],
+                                      q_index) *
+                 fe_values.shape_grad(lexicographic_to_hierarchic_numbering[j],
+                                      q_index) *
+                 fe_values.JxW(q_index));
+            }
+
+      return {mass_matrix_reference, derivative_matrix_reference, false};
+    }
+  } // namespace internal
+
+
+
+  template <int dim, typename Number>
+  std::pair<std::array<FullMatrix<Number>, dim>,
+            std::array<FullMatrix<Number>, dim>>
+  create_laplace_tensor_product_matrix(
+    const FiniteElement<1> &                            fe,
+    const Quadrature<1> &                               quadrature,
+    const dealii::ndarray<LaplaceBoundaryType, dim, 2> &bids,
+    const dealii::ndarray<double, dim, 3> &             cell_extend,
```

Comment:
```suggestion
    const dealii::ndarray<double, dim, 3> &             cell_extent,
```
extend = verb, extent = noun